scyllacloud/hostdialer.go

```diff
@@ -96,8 +96,22 @@ func (s *SniHostDialer) dialInitialContactPoint(ctx context.Context, host *gocql
 		return nil, fmt.Errorf("can't get root CA from configuration: %w", err)
 	}
 
-	return s.connect(ctx, s.dialer, host.HostnameAndPort(), &tls.Config{
-		ServerName:         host.Hostname(),
+	if len(s.connConfig.CurrentContext) == 0 {
+		return nil, fmt.Errorf("current context is empty")
+	}
+
+	contextConf, ok := s.connConfig.Contexts[s.connConfig.CurrentContext]
```

Comment:
it's the same as in GetClientCertificate - can you extract it to a method?

Reply:
done